crates/chain-state/src/in_memory.rs

```diff
@@ -455,24 +455,24 @@ impl BlockState {
             .unwrap_or_default()
     }
 
-    /// Returns a vector of parent `BlockStates` starting from the oldest one.
+    /// Returns a vector of parent `BlockStates` starting from the newest one.
     pub fn parent_state_chain(&self) -> Vec<&Self> {
         let mut parents = Vec::new();
         let mut current = self.parent.as_deref();
 
         while let Some(parent) = current {
-            parents.insert(0, parent);
+            parents.push(parent);
             current = parent.parent.as_deref();
         }
 
         parents
     }
 
     /// Returns a vector of `BlockStates` representing the entire in memory chain,
-    /// including self as the last element.
+    /// including self as the first element.
     pub fn chain(&self) -> Vec<&Self> {
         let mut chain = self.parent_state_chain();
-        chain.push(self);
+        chain.insert(0, self);
         chain
```

Comment:
one improvement we can make here or in a followup is adding another function that is 

`fn append_parent_chain(&self, chain: &mut Vec)` so we don't need to insert0 here which needs to shift the entire chain

Reply:
added here https://github.com/paradigmxyz/reth/pull/9826/commits/7236f14b188179b059bee576508ad7f9313fb0a4 ptal